src/repositories/CourseRepository.js

```diff
@@ -11,7 +11,11 @@ exports.findCourseAndPopulate = (query, origin, populateVirtual = false) => Cour
       : 'misc'
   )
   .populate([
-    { path: 'companies', select: 'name' },
+    {
+      path: 'companies',
+      select: 'name',
+      ...!query.company && { populate: { path: 'holding', populate: { path: 'holding', select: 'name' } } },
```

Comment:
Dans tes tests : quand company est dans la query, que penses-tu de verifier que course a les bon champs populés et notamment que `holding` n'y est pas) ?

Reply:
tu veux retirer le stub du coup ? ça veut dire qu'il faut revoir tous les tests de 'list'..

Reply:
Nop je parle dans les tests d'integrations. Dans le test `should get trainer\'s course (ops mobile)` tu verifies que les champs de la formations sont bien populés. Je me dis que dans le test "should get courses for a specific company (ops webapp) " oû on passe `company` dans la query, on pourrait verifier que course a les bons champs.

Reply:
D'ailleurs dans cette methode on passe plutot `query.companies`  que `query.company` non ? (Donc la condition devrait etre `query.companies` plutot non ?)

Reply:
done